src/main/java/hudson/plugins/sonar/client/SQProjectResolver.java

```diff
@@ -77,6 +87,34 @@ public ProjectInformation get(String projectUrl, String installationName) {
     }
   }
 
+  private void getCETask(WsClient wsClient, ProjectInformation projectInfo, String ceTaskId, Float version) throws Exception {
+    if (version < 5.2f || ceTaskId == null) {
+      return;
+    }
+
+    CETask ceTask = wsClient.getCETask(ceTaskId);
+    projectInfo.setCeStatus(ceTask.getStatus());
+    projectInfo.setCeUrl(ceTask.getUrl());
+
+    if (ceTask.getComponentName() != null) {
+      projectInfo.setName(ceTask.getComponentName());
+    }
+  }
+
+  private static void getQualityGate(WsClient client, ProjectInformation proj, String projectKey, float version) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00112)